core/objects.go

```diff
@@ -398,6 +398,21 @@ func (authz *Authorization) FindChallenge(challengeID int64) int {
 	return -1
 }
 
+// SolvedBy will look through the Authorizations challenges, returning a pointer
+// to the *first* challenge it finds with Status: valid, or nil if no challenge
+// is valid.
+func (authz *Authorization) SolvedBy() *Challenge {
```

Comment:
I think the only thing we ever use from the challenge is the type, there is no real perf reason to just return that since we are just passing around pointers but since we don't actually need anything else from the challenge any reason not to?

Reply:
I thought this might be more general & future-proof but without a concrete use-case I'm also happy to just make it return the challenge type and call it a day.

Reply:
Addressed with 62ad5ac